### PR TITLE
[Task #80844] Improve search function to be able to apply on filtered…

### PR DESCRIPTION
**Pull request by minhsieu9102003**
ready 

https://edu-redmine.sun-asterisk.vn/issues/80844

### CODE REVIEW

**Review by tuan-bach123 — CHANGES_REQUESTED** [review]
Doi nhieu file qua r a

File: ecommerce-app/app/Http/Controllers/CartController.php [hunk 1]
```diff
@@ -48,30 +46,17 @@ public function store(): ShoppingCart
...
+            $order_items = $cart->products()->orderBy("id", "desc")->get()->all();
 
-            return view("cart", ["cart" => $cart, "cart_items" => $cart_items]);
+            return view("cart.show", ["cart" => $cart, "order_items" => $order_items]);
         } else {
             $cart = $this->store();
-            $cart_items = $cart->shoppingCartItems()->orderBy("id", "desc")->get()->all();
-
-            return view("cart", ["cart" => $cart, "cart_items" => $cart_items]);
+            return view("cart.show", ["cart" => $cart, "order_items" => null]);
```
Comment (tuan-bach123) [inline ecommerce-app/app/Http/Controllers/CartController.php]: Ko co view nao la cart.show dau

File: ecommerce-app/routes/web.php [hunk 2]
```diff
@@ -1,50 +1,37 @@
...
     Route::get('/profile', [ProfileController::class, 'edit'])->name('profile.edit');
     Route::patch('/profile', [ProfileController::class, 'update'])->name('profile.update');
     Route::delete('/profile', [ProfileController::class, 'destroy'])->name('profile.destroy');
-
-    Route::get('/order-details', [OrderDetailController::class, 'index'])->name('order-details.index');
-    Route::get('/order-details/{orderDetail}', [OrderDetailController::class, 'show'])->name('order-details.show');
-    
-    Route::get('/cart', [CartController::class, 'show'])->name('cart.show');
-    Route::post('/cart/add', [CartController::class, 'update'])->name('cart.update');
-    Route::delete('/cart/removeItem', [CartController::class, 'removeItem'])->name('cart.removeItem');
```
Comment (tuan-bach123) [inline ecommerce-app/routes/web.php]: check lai xem a van xoa het route cart cua e nay

**Comment by sonnguyendev157** [comment]
close plz

**Review by tuan-bach123 — APPROVED** [review]

**Review by notverygood135 — APPROVED** [review]

**Review by sonnguyendev157 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**